File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.py
class Solution:
    def isSortedColumn(self, col: int, strs: list[str]) -> bool:
        for i in range(1, len(strs)):
            if strs[i][col] < strs[i - 1][col]:
                return False
        return True

    def minDeletionSize(self, strs: list[str]) -> int:
        count = 0
        for i in range(len(strs[0])):
            if not self.isSortedColumn(i, strs):
                count += 1
        return count

File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int res{0};
        auto row{strs.size()}, col{strs[0].size()};
        
        for(int j=0; j<col; j++)
        {
            for(int i=0; i<row-1; i++)
            {
                if(strs[i][j]>strs[i+1][j])
                {
                    res++;
                    break;
                }
            }
        }
        
        return res;
    }
};